### ISSUE TITLE
Allow custom font to be read and set during runtime

### ISSUE THREAD

**Issue by zenith6495** [dotnet/maui#22220]
### Description

Hi .NET MAUI team,

Currently, custom font must be pre-registered in Resources/Fonts folder MauiProgram.cs file in order to use it. However, it would be great if you can allow custom font file to be read to data stream and used by the controls during runtime. 

For instance, currently controls can only specify string for **FontFamily** property which points to the name pre-registered in MauiProgram.cs, it would be great if it accepts byte array too so that custom font can be read as byte array and pass to the controls.

### Public API Changes

<Image>
     <Image.Source>
         <FontImageSource Glyph="&#65;"
                          FontFamily="{Binding SelectedFont}"    <- **besides string, please consider accept byteArray of the font file** 
                          Size="100" />
     </Image.Source>
 </Image>

### Intended Use-Case

The purpose of this feature request is because the application I'm developing requires rendering all font glyphs for ALL installed fonts in windows PC. Thus, pre-register of fonts in Resources/Fonts folder is not an option as there are many fonts in users' PC. 

I sincerely hope this can be consider from you. Thank you!

**Comment by jfversluis** [comment]
I don't think this is possible, at least no on all platforms. I'm pretty sure on iOS you need to declare and load fonts upfront and they need an entry in the `info.plist`. Because of that, chances of this being actually implemented in .NET MAUI are probably pretty slim.

**Comment by raV720** [comment]
It is possible on Android, IOS, WinUi. I did it using reflection and Microsoft.Maui.FontRegistrar so for sure it is possible for MAUI to expose such functionality.
